backend/staticfiles/synthesis/lib/inputs.py

```diff
@@ -17,6 +17,56 @@ class Inputs:
     def __init__(self, input_data) -> None:
         self.inputs = input_data
 
+    def read(self, name):
+        if self.inputs.get(name) is None:
+            raise InvalidInputNameException(f"{name} is not declared in inputs")
+
+        if self.inputs[name].get("created", False):
+            # Do this if read wire has been created
+            # Read data from the different buffers of the SHM Objects
+            dim = create_ndbuffer((1,), np.int64, self.inputs[name]["dim"].buf)[:][0]
+            shape = create_ndbuffer((dim,), np.int64, self.inputs[name]["shape"].buf)
+            type = create_ndbuffer((1,), '<U6', self.inputs[name]["type"].buf)[:][0]
+            data = create_ndbuffer(shape, type, self.inputs[name]["data"].buf)
+        else:
+            # Do this if read wire has not been created
+            # Create required wires(SHM Objects) to read data from the outputs
+            wire_name = self.inputs[name]["wire"]
+            data_wire = create_readonly_wire(wire_name)
+            dim_wire = create_readonly_wire(wire_name + "_dim")
+            shape_wire = create_readonly_wire(wire_name + "_shape")
+            type_wire = create_readonly_wire(wire_name + "_type")
```

Comment:
Can you add '_type' in [clean_shared_memory](https://github.com/JdeRobot/VisualCircuit/blob/master/backend/staticfiles/synthesis/main.py#L19). Its just a fallback in case something is not cleaned.

Reply:
Yep, added :+1: